### DISCUSSION TITLE
Failed to execute 'replaceState' on 'History': A history state object with URL on google web cache

### DISCUSSION THREAD

**Discussion by kamilors** [vercel/next.js#13719] [Help]
# Bug report

Getting 'replaceState' error could not be executed on google cache page. 
Other javascript operations are not executed after the error.

- Version of Next.js: 9.4.4

example: 

test url on chrome: cache:https://www.tiktok.com/en 

http://webcache.googleusercontent.com/search?sxsrf=ALeKk0063qZ_PJbXw3Z1GxleT-FfXnzXCQ%3A1591192988231&ei=nK3XXoPaDYmJk74Pruer6AM&q=cache%3Ahttps%3A%2F%2Fwww.tiktok.com%2Fen&oq=cache%3Ahttps%3A%2F%2Fwww.tiktok.com%2Fen&gs_lcp=CgZwc3ktYWIQAzoECAAQR1C5FliSF2DFGGgAcAF4AIABjAGIAZYCkgEDMC4ymAEAoAEBqgEHZ3dzLXdpeg&sclient=psy-ab&ved=0ahUKEwiDmNTm5-XpAhWJxMQBHa7zCj0Q4dUDCAw&uact=5

`main-8235acc2b9f1a226721e.js:1 Failed to execute 'replaceState' on 'History': A history state object with URL 'https://www.tiktok.com/search?sxsrf=ALeKk0063qZ_PJbXw3Z1GxleT-FfXnzXCQ%3A1591192988231&ei=nK3XXoPaDYmJk74Pruer6AM&q=cache%3Ahttps%3A%2F%2Fwww.tiktok.com%2Fen&oq=cache%3Ahttps%3A%2F%2Fwww.tiktok.com%2Fen&gs_lcp=CgZwc3ktYWIQAzoECAAQR1C5FliSF2DFGGgAcAF4AIABjAGIAZYCkgEDMC4ymAEAoAEBqgEHZ3dzLXdpeg&sclient=psy-ab&ved=0ahUKEwiDmNTm5-XpAhWJxMQBHa7zCj0Q4dUDCAw&uact=5' cannot be created in a document with origin 'http://webcache.googleusercontent.com' and URL 'http://webcache.googleusercontent.com/search?sxsrf=ALeKk0063qZ_PJbXw3Z1GxleT-FfXnzXCQ%3A1591192988231&ei=nK3XXoPaDYmJk74Pruer6AM&q=cache%3Ahttps%3A%2F%2Fwww.tiktok.com%2Fen&oq=cache%3Ahttps%3A%2F%2Fwww.tiktok.com%2Fen&gs_lcp=CgZwc3ktYWIQAzoECAAQR1C5FliSF2DFGGgAcAF4AIABjAGIAZYCkgEDMC4ymAEAoAEBqgEHZ3dzLXdpeg&sclient=psy-ab&ved=0ahUKEwiDmNTm5-XpAhWJxMQBHa7zCj0Q4dUDCAw&uact=5'.
Error: Failed to execute 'replaceState' on 'History': A history state object with URL 'https://www.tiktok.com/search?sxsrf=ALeKk0063qZ_PJbXw3Z1GxleT-FfXnzXCQ%3A1591192988231&ei=nK3XXoPaDYmJk74Pruer6AM&q=cache%3Ahttps%3A%2F%2Fwww.tiktok.com%2Fen&oq=cache%3Ahttps%3A%2F%2Fwww.tiktok.com%2Fen&gs_lcp=CgZwc3ktYWIQAzoECAAQR1C5FliSF2DFGGgAcAF4AIABjAGIAZYCkgEDMC4ymAEAoAEBqgEHZ3dzLXdpeg&sclient=psy-ab&ved=0ahUKEwiDmNTm5-XpAhWJxMQBHa7zCj0Q4dUDCAw&uact=5' cannot be created in a document with origin 'http://webcache.googleusercontent.com' and URL 'http://webcache.googleusercontent.com/search?sxsrf=ALeKk0063qZ_PJbXw3Z1GxleT-FfXnzXCQ%3A1591192988231&ei=nK3XXoPaDYmJk74Pruer6AM&q=cache%3Ahttps%3A%2F%2Fwww.tiktok.com%2Fen&oq=cache%3Ahttps%3A%2F%2Fwww.tiktok.com%2Fen&gs_lcp=CgZwc3ktYWIQAzoECAAQR1C5FliSF2DFGGgAcAF4AIABjAGIAZYCkgEDMC4ymAEAoAEBqgEHZ3dzLXdpeg&sclient=psy-ab&ved=0ahUKEwiDmNTm5-XpAhWJxMQBHa7zCj0Q4dUDCAw&uact=5'.
    at t.value (https://s16.tiktokcdn.com/tiktok/falcon/_next/static/chunks/52552abc5a244128f475a2434072c9895be5b417.e911047b168c9ecda2b0.js:1:15810)
    at new t (https://s16.tiktokcdn.com/tiktok/falcon/_next/static/chunks/52552abc5a244128f475a2434072c9895be5b417.e911047b168c9ecda2b0.js:1:13385)
    at a (https://s16.tiktokcdn.com/tiktok/falcon/_next/static/chunks/52552abc5a244128f475a2434072c9895be5b417.e911047b168c9ecda2b0.js:1:22036)
    at e.createRouter (https://s16.tiktokcdn.com/tiktok/falcon/_next/static/chunks/52552abc5a244128f475a2434072c9895be5b417.e911047b168c9ecda2b0.js:1:21211)
    at https://s16.tiktokcdn.com/tiktok/falcon/_next/static/runtime/main-8235acc2b9f1a226721e.js:1:14802
    at u (https://s16.tiktokcdn.com/tiktok/falcon/_next/static/chunks/52552abc5a244128f475a2434072c9895be5b417.e911047b168c9ecda2b0.js:1:2714)
    at Generator._invoke (https://s16.tiktokcdn.com/tiktok/falcon/_next/static/chunks/52552abc5a244128f475a2434072c9895be5b417.e911047b168c9ecda2b0.js:1:2467)
    at Generator.forEach.t.<computed> [as next] (https://s16.tiktokcdn.com/tiktok/falcon/_next/static/chunks/52552abc5a244128f475a2434072c9895be5b417.e911047b168c9ecda2b0.js:1:3071)
    at u (https://s16.tiktokcdn.com/tiktok/falcon/_next/static/chunks/52552abc5a244128f475a2434072c9895be5b417.e911047b168c9ecda2b0.js:1:2714)
    at n (https://s16.tiktokcdn.com/tiktok/falcon/_next/static/chunks/52552abc5a244128f475a2434072c9895be5b417.e911047b168c9ecda2b0.js:1:3205)
(anonymous) @ main-8235acc2b9f1a226721e.js:1`

**Comment by kamilors** [comment]
workaround:

use cdn for all resources and remove the base tag added by google.

JQuery : $('base').remove();

**Reply by hoangtrieukd** [reply]
Hi @kamilors i had same your issue. Your issue was fixed? Could you explain more how you fix it?

**Reply by demirelrecep** [reply]
@hoangtrieukd  hi, i had the same issue. Did you fix it?

**Reply by hoangtrieukd** [reply]
@demirelrecep  I pushed this code to html and it worked.
`
 <script
            dangerouslySetInnerHTML={{
              __html: `
              if (window.location.origin.includes("https://webcache.googleusercontent.com")) {
                window.OhistoryReplaceState = window.history["replaceState"];
                window.history["replaceState"] = (...args)=> {
                  try {
                      return window.OhistoryReplaceState.apply(window.history, args)
                  } catch (e) {
                      console.log(e)
                  }
                };
  
                window.OhistoryPushState = window.history["pushState"];
                window.history["pushState"] = (...args)=> {
                  try {
                      return window.OhistoryPushState.apply(window.history, args)
                  } catch (e) {
                      console.log(e)
                  }
                };
              }
              `,
            }}
          />
`

**Reply by demirelrecep** [reply]
@hoangtrieukd thank you, i will try this.

**Reply by FatemaDholkawala** [reply]
@demirelrecep  Hi, did the solution work for you?

**Reply by Husain010** [reply]
Is this the only way to handle this? Any better solutions to this? Also why is it caused?
